server/src/main/java/org/elasticsearch/rest/RestResponse.java

```diff
@@ -131,6 +131,9 @@ public RestResponse(RestChannel channel, RestStatus status, Exception e) throws
                 SUPPRESSED_ERROR_LOGGER.warn(messageSupplier, e);
             }
         }
+        if (params.paramAsBoolean("error_trace", false) && status != RestStatus.UNAUTHORIZED) {
```

Comment:
I know you are just rearranging code for this case, but (if you understand it) could you add a code comment of what this is for and what it is doing? It's not clear to me.

Reply:
I added comments